src/test/java/com/google/api/codegen/protoannotations/testdata/library_v2_gapic.yaml

```diff
@@ -0,0 +1,693 @@
+type: com.google.api.codegen.ConfigProto
+config_schema_version: 2.0.0
+language_settings:
+  java:
+    package_name: com.google.example.library.v1
+    interface_names:
+      google.example.library.v1.LibraryService: Library
+    release_level: GA
+  python:
+    package_name: google.cloud.example.library_v1.gapic
+  go:
+    # Intentionally changed from the gRPC package name.
+    # See: https://github.com/googleapis/toolkit/issues/320
+    package_name: cloud.google.com/go/library/apiv1
+    domain_layer_location: cloud.google.com/go/library
+    interface_names:
+      google.example.library.v1.LibraryService: LibClient
+  ruby:
+    # Note that package_name for ruby is intentionally changed
+    # from the gRPC package name of the API to test the generated
+    # code. See: https://github.com/googleapis/toolkit/issues/309
+    package_name: Library::V1
+  # Use default PHP package name.
+  nodejs:
+    package_name: library.v1
+    domain_layer_location: google-cloud
+  csharp:
+    package_name: Google.Example.Library.V1
+interfaces:
+  - name: google.example.library.v1.MyProto
+    retry_codes_def:
+      - name: idempotent
+        retry_codes:
+          - UNAVAILABLE
+          - DEADLINE_EXCEEDED
+      - name: non_idempotent
+        retry_codes: []
+    methods:
+      - name: MyMethod
+        retry_codes_name: non_idempotent
+        retry_params_name: default
+        timeout_millis: 1000
+  - name: google.example.library.v1.LibraryService
+    lang_doc:
+      java: >
+        Also see this awesome doc there!
+        and there!
+        and everywhere!
+      go: >
+        Write more docs here as an offering to the Great Gopher
+    retry_codes_def:
+      - name: idempotent
+        retry_codes:
+          - UNAVAILABLE
+          - DEADLINE_EXCEEDED
+      - name: non_idempotent
+        retry_codes: []
+    retry_params_def:
+      - name: default
+        initial_retry_delay_millis: 100
+        retry_delay_multiplier: 1.2
+        max_retry_delay_millis: 1000
+        initial_rpc_timeout_millis: 300
+        rpc_timeout_multiplier: 1.3
+        max_rpc_timeout_millis: 3000
+        total_timeout_millis: 30000
+    collections:
+      - entity_name: Book
+        language_overrides:
+          - language: java
+            entity_name: shelf_book
+      - entity_name: Project
+        language_overrides:
+          - language: csharp
+            common_resource_name: Google.Api.Gax.ResourceNames.ProjectName
+          - language: java
+            common_resource_name: com.google.cloud.ProjectName
+    smoke_test:
+      method: UpdateBook
+      init_fields:
+        - name%shelf_id="testShelf-$RANDOM"
+        - name%book_id=$PROJECT_ID
+        - book . rating=GOOD
+    methods:
+      - name: CreateShelf
+        retry_codes_name: non_idempotent
```

Comment:
Can these be inferred from http annotation? We should have a mix of most being inferred, but some being overridden. WDYT?

Reply:
Ahh I got lazy
Removed the retry codes names that are the same as the inferred value.